### ISSUE TITLE
How to Implement A Semi-Supervised Model

### ISSUE THREAD

**Issue by 3rduncle** [fchollet/keras#2380]
I want to implement a semi-supervise model in this paper http://arxiv.org/abs/1603.06318
I try with steps below:
1. In my loss function, I pass a symbol x, representing unlabeled data, into the network to get the tensor of whole net.
2. computation of such tensor is combined with the original loss function. then the affected gradient will be computed automatically and go backward to tune the weight.
3. In training, I assign some unlabeled data  to symbol x.
code like this:

``` py
def semi_loss(y_true, y_pred):
    labeled_loss = K.mean(K.binary_crossentropy(y_pred, y_true), axis=-1)
    unlabeled_x = T.matrix('unlabeled_x')
    unlabeled_tensor = net(unlabeled_x)
    unlabeled_loss = regulation(unlabeled_tensor)
    return 0.9 * labeled_loss + 0.1 * unlabeled_loss
```

But I don't know how to get such tensor about `unlabeled_x`.
Any suggestions?

**Comment by carlthome** [comment]
I had a similar question with how to get additional parameters into the loss function. https://github.com/fchollet/keras/issues/2121 and @rpinsler suggested his very workable solution in https://github.com/fchollet/keras/issues/1061 - perhaps that would work for you?

**Comment by zero76114** [comment]
@3rduncle : Can you share your full code?
